src/lib/Microsoft.Health.Events/EventHubProcessor/BaseEventProcessor.cs

```diff
@@ -104,5 +104,22 @@ protected virtual async Task ProcessInitializingHandler(PartitionInitializingEve
                 EventHubExceptionProcessor.ProcessException(ex, Logger, errorMetricName: EventHubErrorCode.EventHubPartitionInitFailed.ToString());
             }
         }
+
+        protected virtual void HandleOwnershipFailure(AggregateException ex)
```

Comment:
The event hub code seems to always throw an aggregate exception for the ownership failure: https://github.com/Azure/azure-sdk-for-net/blob/ae550cb5d4be459e61d0f42f46c00033b0fc9a83/sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/EventProcessor.cs#L1266

As far as I know this is the only exception we've seen at startup. We could have a catch-all for any other Exception and we could periodically retry just like we are doing here for the ownership failure, but it is sort of unclear if that is appropriate. In some cases it may be better to throw the exception and crash and see if the restart fixes the issue.

Reply:
updated to use the LINQ suggestion